src/main/java/com/linkedin/xinfra/monitor/services/ConsumeService.java

```diff
@@ -211,6 +214,10 @@ public void onComplete(Map<TopicPartition, OffsetAndMetadata> topicPartitionOffs
       }
     }
     /* end of consume() while loop */
+    LOG.info("{}/ConsumeService/Consumer closing.", _name);
+    _baseConsumer.close();
+    LOG.info("{}/ConsumeService/Consumer stopped.", _name);
+    Thread.sleep(1000);
```

Comment:
`Thread.sleep` directly is a big bad smell in Java concurrent code. What's the purpose of this?